lib/preboots/scheduler.js

```diff
@@ -0,0 +1,162 @@
+const EE = require('events');
+const async = require('async');
+const once = require('one-time');
+const Lookup = require('../lookup');
+const nsqStream = require('nsq-stream');
+const from = require('from2');
+const util = require('util');
+
+const assign = Object.assign;
+
+util.inherits(Scheduler, EE);
+
+/**
+ * Scheduler
+ * @public
+ * @constructor
+ * @param {Object} options - Configuration object
+ * @param {Object} options.log - Logger
+ * @param {Object} options.nsq - NSQ object
+ * @param {String} options.topic - NSQ topic to dispatch onto
+ * @param {Object} options.models - Data models
+ * @param {Number} [options.concurrency=5] - Number of jobs to run concurrently
+ * @param {Number} [options.interval=3600000] - Default interval between catch-up jobs in ms
+ */
+function Scheduler(options) {
+  EE.call(this);
+  this.log = options.log;
+  this.nsq = options.nsq;
+  this.topic = options.topic;
+  this.models = options.models;
+  this.conc = options.concurrency || 5;
+  this.defaultInterval = options.interval || 60 * 60 * 1000;
+
+  this.intervals = new Map();
+}
+
+/**
+ * Set an interval to schedule catch up jobs for the given env
+ * @public
+ * @function setInterval
+ * @param {String} env - Environment
+ * @param {Number} [time] - milliseconds of interval
+ */
+Scheduler.prototype.setInterval = function interval(env, time) {
+  time = time || this.defaultInterval;
+  const intervalId = setInterval(() => {
+    this.emit('schedule');
+    this.schedule(env, (err, counts) => {
+      if (err) return this.log.error('Failed to schedule jobs for %s', env, {
+        error: err.message,
+        stack: err.stack
+      });
+      this.emit('scheduled', err, counts);
```

Comment:
You will never emit `'scheduled'` with an `err` due to the `return` on the previous line.